src/amo/sagas/userAbuseReports.js

```diff
@@ -0,0 +1,49 @@
+// Disabled because of
+// https://github.com/benmosher/eslint-plugin-import/issues/793
+/* eslint-disable import/order */
+import { call, put, select, takeLatest } from 'redux-saga/effects';
+/* eslint-enable import/order */
+
+import {
+  SEND_USER_ABUSE_REPORT,
+  loadUserAbuseReport,
+} from 'amo/reducers/userAbuseReports';
+import { reportUser as reportUserApi } from 'core/api/abuse';
+import log from 'core/logger';
+import { createErrorHandler, getState } from 'core/sagas/utils';
+
+
+export function* reportUser({
+  payload: {
+    errorHandlerId,
+    message,
+    user,
+  },
+}) {
+  const errorHandler = createErrorHandler(errorHandlerId);
+
+  yield put(errorHandler.createClearingAction());
+
+  try {
+    const state = yield select(getState);
+
+    const response = yield call(reportUserApi, {
+      api: state.api,
+      message,
+      user,
+    });
+
+    yield put(loadUserAbuseReport({
+      message: response.message,
+      reporter: response.reporter,
+      user: response.user,
+    }));
+  } catch (error) {
+    log.warn(`Reporting user for abuse failed: ${error}`);
+    yield put(errorHandler.createErrorAction(error));
```

Comment:
In addition to dispatching this action, I think it helps to dispatch an `abortReportUser` action that can reset all the state. The alternative is for the component to check `errorHandler.hasError()` every step of the way but that's hard to get right. The render states for `ReportUserAbuse` are already complex and I'm not sure they fully handle the `errorHandler.hasError()` case.